cf-monitord/env_monitor.c

```diff
@@ -269,7 +269,7 @@ void MonitorStartServer(EvalContext *ctx, const Policy *policy)
         Bundle *bp = PolicyAppendBundle(monitor_cfengine_policy, NamespaceDefault(), "monitor_cfengine_bundle", "agent", NULL, NULL);
         PromiseType *tp = BundleAppendPromiseType(bp, "monitor_cfengine");
 
-        pp = PromiseTypeAppendPromise(tp, "the monitor daemon", (Rval) { NULL, RVAL_TYPE_NOPROMISEE }, NULL, NULL);
+        pp = PromiseTypeAppendPromise(tp, "the monitor daemon", RvalNULL(), NULL, NULL);
```

Comment:
All these could go in a preparatory commit, together with other stylistic stuff you have like whitespace, introducing new functions and using them in various places etc. That preparatory commit should would not change the actual  functionality of CFEngine, but will make the commit adding the actual feature really really small, which is nice. :-)